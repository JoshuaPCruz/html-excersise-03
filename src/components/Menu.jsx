import React from 'react';
import '../css/components/ButtonMenu.css'
import '../css/components/Menu.css'

const Menu = () => {
  return (
    <ul className="Menu">
      <li className="Button-Menu-Active">Inicio</li>
      <li className="Button-Menu">Acerca</li>
      <li className="Button-Menu">Servicios</li>
      <li className="Button-Menu">Productos</li>
      <li className="Button-Menu">Contacto</li>
    </ul>
  );
};

export default Menu;