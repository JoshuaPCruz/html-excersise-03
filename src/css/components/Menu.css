.Menu {
  display: flex;
}

.Menu li {
  display: inline-block;
  margin-left: 35px;
}

.Menu li:first-child {
  margin-left: 0;
}