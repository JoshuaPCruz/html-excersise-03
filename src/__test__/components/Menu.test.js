import React from 'react';
import { shallow } from 'enzyme';
import { create } from 'react-test-renderer';
import Menu from '../../components/Menu';

describe('<Menu />', () => {
  const menu = shallow(<Menu />);
  test('Render del componente Menu', () => {
    expect(menu.length).toEqual(1);
  });

  test('Render 5 listed elements in the menu', () => {
    expect(menu.find('ul').children()).toHaveLength(5);
  });
});

describe('Menu Snapshot', () => {
  test('Comprobar la UI del componente Menu', () => {
    const menu = create(<Menu />);
    expect(menu.toJSON()).toMatchSnapshot();
  });
});