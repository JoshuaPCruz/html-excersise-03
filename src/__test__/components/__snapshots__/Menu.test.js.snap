// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu Snapshot Comprobar la UI del componente Menu 1`] = `
<ul
  className="Menu"
>
  <li
    className="Button-Menu-Active"
  >
    Inicio
  </li>
  <li
    className="Button-Menu"
  >
    Acerca
  </li>
  <li
    className="Button-Menu"
  >
    Servicios
  </li>
  <li
    className="Button-Menu"
  >
    Productos
  </li>
  <li
    className="Button-Menu"
  >
    Contacto
  </li>
</ul>
`;
